cfme/infrastructure/virtual_machines.py

```diff
@@ -86,46 +86,110 @@
 retire_remove_button = "//span[@id='remove_button']/a/img"
 
 match_page = partial(match_location, controller='vm_infra', title='Virtual Machines')
-vm_templates_tree = partial(accordion.tree, "VMs & Templates")
-vms_tree = partial(accordion.tree, "VMs")
-templates_tree = partial(accordion.tree, "Templates")
-
-
-def reset_page():
-    tb.select("Grid View")
-    from cfme.web_ui import paginator
-    if sel.is_displayed(search_box.search_field):
-        search.ensure_normal_search_empty()
-    if paginator.page_controls_exist():
-        # paginator.results_per_page(1000)
-        paginator.check_all()
-        paginator.uncheck_all()
 
 
 drift_table = CheckboxTable("//th[normalize-space(.)='Timestamp']/ancestor::table[1]")
 
 
-class InfraVmDetailsToolbar(View):
-    configuration = Dropdown("Configuration")
-    policy = Dropdown("Policy")
-    lifecycle = Dropdown("Lifecycle")
+class InfraVmDetailsToolbar(VMToolbar):
```

Comment:
VMToolbar is written for the 'All' view, and has a dropdown for the download button.

For the details toolbar, you'll want to either add a common view for details toolbar or add all the widgets here and build it off of View.

cloud.instance.InstanceDetailsToolbar inherits View directly and implements all its widgets, mostly because the 'Power' button title is different between instances and infra VM's.